src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -583,6 +583,12 @@ TER DirectStepI::check (StrandContext const& ctx) const
         return temBAD_PATH;
     }
 
+    if (src_ == dst_)
+    {
+        JLOG (j_.debug()) << "DirectStepI: same src and dst.";
+        return temBAD_PATH;
```

Comment:
One question about this change that just hit me.  Is this too late in transaction processing to be returning a `tem`?  Do we need to be returning a `tec`?